packages/next/src/server/lib/patch-fetch.ts

```diff
@@ -191,10 +224,14 @@ export function patchFetch({
   const { DynamicServerError } = serverHooks
   const originFetch: typeof fetch = (globalThis as any)._nextOriginalFetch
 
-  globalThis.fetch = async (
+  async function patchedFetch(
     input: RequestInfo | URL,
     init: RequestInit | undefined
-  ) => {
+  ) {
+    const tracingError = new Error()
```

Comment:
can we create it lazily?

Reply:
comments explaining what this does would be nice

Reply:
Updated the comment to explain more. the error has to be created here to contain the call stack. But we don't need to throw it always. We only replace the stack trace when necessary